src/plugins/intel_cpu/src/nodes/kernels/stack_allocator.hpp

```diff
@@ -0,0 +1,477 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <math.h>
+#include <dnnl_extension_utils.h>
+#include <cpu/x64/jit_generator.hpp>
+#include <stack>
+
+namespace ov {
+namespace intel_cpu {
+
+using namespace dnnl::impl;
+using namespace dnnl::impl::cpu;
+
+class StackAllocator final {
+public:
+    class Transaction;
+    class Address;
+    template<typename TReg>
+    class Reg;
+
+    StackAllocator(x64::jit_generator& code_gen)
+        : StackAllocator{code_gen, code_gen.rbp} {
+    }
+
+    StackAllocator(x64::jit_generator& code_gen, const Xbyak::Reg& bp)
+        : StackAllocator{code_gen, bp, 1} {
+    }
+
+    StackAllocator(x64::jit_generator& code_gen, const size_t alignment)
+        : StackAllocator{code_gen, code_gen.rbp, alignment} {
+    }
+
+    StackAllocator(x64::jit_generator& code_gen,
+                   const Xbyak::Reg& bp,
+                   const size_t alignment)
+        : code_generator{code_gen}
+        , base_pointer{Xbyak::Reg64{bp.getIdx()}}
+        , alignment{alignment} {
+        checkUnique(true);
+        alignStack(true);
+        code_generator.mov(base_pointer, code_generator.rsp);
```

Comment:
When it comes to `StackAllocator::Address` it is all right. I'm talking about the `StackAllocator` itself (the comment placed under the `StackAllocator` constructor). When we construct and destruct the object it call some code generation methods to the `code_generator`. In case of constructor we know the place and the moment of time when code generation method is called, but in case of destructor it is not that obvious. We can allocate a `StackAllocator` object on heap, hand over the lifetime control to `shared_ptr` and, in such a scenario the moment of calling the destructor is depends on the reference count. In turn, it leads to a situation when the destructor and, consequently, some code generation are called in wrong place, breaking commands sequence. To avoid such scenario, it is better to decouple instancing and code generation, extracting code generation operations to separate methods.

Reply:
@maxnick 
> When it comes to `StackAllocator::Address` it is all right. I'm talking about the `StackAllocator` itself (the comment placed under the `StackAllocator` constructor). When we construct and destruct the object it call some code generation methods to the `code_generator`. In case of constructor we know the place and the moment of time when code generation method is called, but in case of destructor it is not that obvious. We can allocate a `StackAllocator` object on heap, hand over the lifetime control to `shared_ptr` and, in such a scenario the moment of calling the destructor is depends on the reference count. In turn, it leads to a situation when the destructor and, consequently, some code generation are called in wrong place, breaking commands sequence. To avoid such scenario, it is better to decouple instancing and code generation, extracting code generation operations to separate methods.

This issue is already fixed. `StackAllocator` has special method `release`.
See it here:
https://github.com/openvinotoolkit/openvino/pull/13132/files#diff-ad5c4db2706d50420066fbb4beedb2fac0532bf2ba5468e6a9dd53eedbbaa6b7R56